javaee/ejb-mdb-activemq/src/main/java/org/wildfly/swarm/ts/javaee/ejb/mdb/activemq/MessagesResource.java

```diff
@@ -0,0 +1,45 @@
+package org.wildfly.swarm.ts.javaee.ejb.mdb.activemq;
+
+import javax.annotation.Resource;
+import javax.inject.Inject;
+import javax.jms.Connection;
+import javax.jms.ConnectionFactory;
+import javax.jms.JMSException;
+import javax.jms.Queue;
+import javax.jms.Session;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+
+@Path("/")
+public class MessagesResource {
+    @Inject
+    private Result result;
+
+    // the ActiveMQ resource adapter doesn't support JMS 2.0 client APIs
+
+    @Resource(lookup = "java:comp/DefaultJMSConnectionFactory")
+    private ConnectionFactory connectionFactory;
+
+    @GET
+    @Produces(MediaType.TEXT_PLAIN)
+    public String receivedMessages() {
+        return String.join("|", result.getItems());
+    }
+
+    @POST
+    @Consumes(MediaType.TEXT_PLAIN)
+    public Response sendMessage(String text) throws JMSException {
+        Connection connection = connectionFactory.createConnection();
```

Comment:
I think this could be improved by using auto-closeable try() {} for both Connection and Session.

Reply:
I don't think try-with-resources would work, as the ActiveMQ JMS client doesn't implement JMS 2.0 (where `Connection` implements `AutoCloseable`), only 1.1 (where it doesn't, not even `Closeable`). So I'd have to go with regular `try`/`catch`/`finally`, which is IMHO too cumbersome for this test :-)

Reply:
Oh ok, I think I didn't pay enough attention to the JMS 2.0 point. Thanks!